### PR TITLE
INTMDB-658: [Terraform] Upgrade to go 1.20

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/provider_test.go [hunk 1]
```diff
@@ -39,56 +39,56 @@ func TestProvider(t *testing.T) {
...
-func testAccPreCheckGPCEnv(t *testing.T) {
+func testAccPreCheckGPCEnv(tb testing.TB) {
 	if os.Getenv("GCP_SERVICE_ACCOUNT_KEY") == "" || os.Getenv("GCP_KEY_VERSION_RESOURCE_ID") == "" {
-		t.Fatal("`GCP_SERVICE_ACCOUNT_KEY` and `GCP_KEY_VERSION_RESOURCE_ID` must be set for acceptance testing")
+		tb.Fatal("`GCP_SERVICE_ACCOUNT_KEY` and `GCP_KEY_VERSION_RESOURCE_ID` must be set for acceptance testing")
 	}
 }
 
-func checkPeeringEnvAWS(t *testing.T) {
+func testCheckPeeringEnvAWS(tb testing.TB) {
```
Comment (gssbzn) [inline mongodbatlas/provider_test.go:73]: were these because of a linter change?

**Review by gssbzn — APPROVED** [review]
LGTM

File: mongodbatlas/provider_test.go (hunk 1 above)
Comment (andreaangiolillo) [reply mongodbatlas/provider_test.go:73]: yes

Comment (andreaangiolillo) [reply mongodbatlas/provider_test.go:73]: Here is the lint action that was failing https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/4863632956/jobs/8671593065

```bash
2023-05-02T16:41:45.6158922Z Running [/home/runner/golangci-lint-1.52.2-linux-amd64/golangci-lint run --out-format=github-actions --timeout 9m0s] in [] ...
2023-05-02T16:44:05.9298193Z ##[error]sloppyTestFuncName: function checkFederatedSettings should be of form TestXXX(t *testing.T) (gocritic)
2023-05-02T16:44:05.9302416Z ##[error]sloppyTestFuncName: function checkLDAP should be of form TestXXX(t *testing.T) (gocritic)
2023-05-02T16:44:05.9304193Z ##[error]sloppyTestFuncName: function checkTeamsIds should be of form TestXXX(t *testing.T) (gocritic)
2023-05-02T16:44:05.9306001Z ##[error]sloppyTestFuncName: function testAccPreCheck looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
2023-05-02T16:44:05.9307626Z ##[error]sloppyTestFuncName: function testAccPreCheckBetaFeatures looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
2023-05-02T16:44:05.9309681Z ##[error]sloppyTestFuncName: function testAccPreCheckGPCEnv looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
2023-05-02T16:44:05.9311304Z ##[error]sloppyTestFuncName: function testAccPreCheckGov looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
2023-05-02T16:44:05.9312893Z ##[error]sloppyTestFuncName: function testCheckAwsEnv looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
2023-05-02T16:44:05.9314850Z ##[error]sloppyTestFuncName: function testCheckEncryptionAtRestEnvAzure looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
2023-05-02T16:44:05.9316305Z ##[error]sloppyTestFuncName: function testCheckPeeringEnvAWS looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
2023-05-02T16:44:05.9317730Z ##[error]sloppyTestFuncName: function testCheckPeeringEnvAzure looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
2023-05-02T16:44:05.9319210Z ##[error]sloppyTestFuncName: function testCheckPeeringEnvGCP looks like a test helper, consider to change 1st param to 'tb testing.TB' (gocritic)
```

Comment (gssbzn) [reply mongodbatlas/provider_test.go:73]: interesting, thanks for sharing

**Comment by andreaangiolillo** [comment]
Hi @martinstibbe 👋 Could you review this PR when you have time, please?

**Review by Zuhairahmed — CHANGES_REQUESTED** [review]
@andreaangiolillo synced with @martinstibbe and we opted to push this one back to to v1.10.1 release to prevent any conflicts with on-going feature updates as part of v1.10.0 release. thank you!

**Comment by andreaangiolillo** [comment]
@Zuhairahmed Should I merge the PR into master instead? Thanks!

**Comment by Zuhairahmed** [comment]
> @Zuhairahmed Should I merge the PR into master instead? Thanks!

Yes please merge into master per https://mongodb.slack.com/archives/C032Q3G3Z28/p1683737491456979

**Review by Zuhairahmed — APPROVED** [review]

**Pull request merged**